### PR TITLE
Fixed IPFS installation

### PR DESCRIPTION
**Pull request by odisseus**

### CODE REVIEW

File: src/core/external.scala [hunk 1]
```diff
@@ -105,7 +105,8 @@ object Ipfs {
       case e: RuntimeException if e.getMessage.contains("Couldn't connect to IPFS daemon") =>
         log.infoWhen(!quiet)(msg"Couldn't connect to IPFS daemon")
         for {
-          ipfs <- IpfsSoftware.installedPath(env, quiet)
+          _    <- IpfsSoftware.installedPath(env, quiet)
+          ipfs =  IpfsSoftware.managedPath
```
Comment (propensive) [inline src/core/external.scala]: This looks wrong... `installedPath` will install `ipfs` only if it's required, then the second line will use the path which assumes IPFS is installed.

File: src/core/external.scala [hunk 2]
```diff
@@ -143,9 +144,9 @@ abstract class Installable(name: ExecName) extends Software(name) {
       _   <- ~log.infoWhen(!quiet)(msg"Downloading $bin...")
       in  <- Http.requestStream(bin, Map[String, String](), "GET", Set())
       _   <- TarGz.extract(in, base)
-      _   <- ~managedPath.setExecutable(true)
+      _   <- managedPath.setExecutable(true)
```
Comment (propensive) [inline src/core/external.scala:147]: With the fix for `TarGz.extract`, this line should be redundant anyway.

File: src/io/path.scala [hunk 3]
```diff
@@ -122,7 +122,7 @@ case class Path(input: String) {
     this
   }
 
-  def setExecutable(exec: Boolean): Try[Unit] = Try(javaFile.setExecutable(exec))
+  def setExecutable(exec: Boolean): Try[Unit] = Try(javaFile.setExecutable(exec)).collect{ case true => () }
```
Comment (propensive) [inline src/io/path.scala]: Maybe we could transform the exception type into something better?

**Review by propensive — COMMENTED** [review]
So, does this make things work? It seems surprising... (and even if it does work, I doubt it will use an existing installation of `ipfs` when it should).

File: src/core/external.scala (hunk 1 above)
Comment (odisseus) [reply src/core/external.scala]: Good point; this must be the cause behind the timeouts I'm getting.

File: src/core/external.scala (hunk 2 above)
Comment (odisseus) [reply src/core/external.scala:147]: I'm not so sure. In an arbitrary distribution archive, the executable files may or may not have the correct permissions, whereas we need to make sure they are set once these files are extracted.

File: src/io/path.scala (hunk 3 above)
Comment (odisseus) [reply src/io/path.scala]: Done

**Pull request merged**